### PR TITLE
Factor out BHEAD and BBODY rules

### PR DESCRIPTION
**Pull request by nc6**
Addresses #258 

This commit splits the block header and body processing into separate rules.

### CODE REVIEW

File: specs/chain/latex/blockchain-spec.tex [hunk 1]
```diff
@@ -824,120 +929,102 @@ \section{Blockchain Extension}
...
+\end{equation*}
+\vspace{20pt}
+\begin{equation*}
   \inference
   {
     \neg\isebb{b} \\
     \fun{dms} ~ \var{ds} \vdash
     {\left(
         \begin{array}{l}
-          \sepoch{s_{last}} \\
```
Comment (dnadales) [inline specs/chain/latex/blockchain-spec.tex:838]: We have one environment too many now

image

**Review by dnadales — APPROVED** [review]
LGTM modulo the extra environment. 

Thanks Nick!

File: specs/chain/latex/blockchain-spec.tex (hunk 1 above)
Comment (nc6) [reply specs/chain/latex/blockchain-spec.tex:838]: Whoops...

**Pull request merged**